### PR TITLE
Add thinking and talking indicators back in

### PR DESCRIPTION
**Pull request by sujayakar**

### CODE REVIEW

File: convex/world.ts [hunk 1]
```diff
@@ -204,11 +205,17 @@ export const sendWorldInput = mutation({
   },
 });
 
+export type ActivePlayer = Doc<'players'> & {
```
Comment (ianmacartney) [inline convex/world.ts:208]: I understand "Active" here to mean "on the map" vs. a human player who has left.

File: convex/world.ts [hunk 2]
```diff
@@ -223,7 +230,26 @@ export const activePlayers = query({
...
+        .withIndex('playerId', (q) =>
+          q.eq('playerId', player._id).eq('status.kind', 'participating'),
+        )
+        .first();
+      if (member) {
+        const indicator = await ctx.db
+          .query('typingIndicator')
+          .withIndex('conversationId', (q) => q.eq('conversationId', member.conversationId))
+          .first();
+        isSpeaking = !!indicator && indicator.typing?.playerId === player._id;
```
Comment (ianmacartney) [inline convex/world.ts:245]: Is there anywhere that's pacing the conversation - so they aren't "speaking" only for the duration of the LLM call, and aren't replying faster than human-speed?

File: data/characters.ts [hunk 3]
```diff
@@ -78,78 +78,6 @@ export const Descriptions = [
       Often it seems like she is close to having a mental breakdown.`,
     plan: 'You want find a way to be happy.',
   },
-
-  // Additional characters (reusing the same sprites):
-  {
-    name: 'Milo',
-    character: 'f5',
-    identity: `Milo is a wanderer, always on the move. He's experienced a lot in his travels,
-        meeting different people and trying different cuisines. He carries a backpack
```
Comment (ianmacartney) [inline data/characters.ts:87]: deleted on purpose? I don't recognize these so am not tied to them. Just thinking about folks who will be merging / rebasing all these changes

File: convex/agent/memory.ts [hunk 4]
```diff
@@ -28,6 +29,9 @@ export async function rememberConversation(
   if (!messages.length) {
     return;
   }
+  const now = Date.now();
+  await ctx.runMutation(selfInternal.startThinking, { agentId, now });
+  await ctx.scheduler.runAfter(ACTION_TIMEOUT, selfInternal.clearThinking, { agentId, since: now });
```
Comment (ianmacartney) [inline convex/agent/memory.ts:34]: I'll wrap this up in a function once there's other ways to "think"

File: convex/agent/memory.ts [hunk 5]
```diff
@@ -157,6 +161,36 @@ export const loadMessages = internalQuery({
...
+  args: {
+    agentId: v.id('agents'),
+    now: v.number(),
+  },
+  handler: async (ctx, args) => {
+    await ctx.db.patch(args.agentId, { isThinking: { since: args.now } });
+  },
+});
+
+export const clearThinking = internalMutation({
```
Comment (ianmacartney) [inline convex/agent/memory.ts:174]: I think we're missing an explicit call to `clearThinking` at the end of the action- as is I suspect it'll always stay thinking for 60s?

File: data/characters.ts (hunk 3 above)
Comment (sujayakar) [reply data/characters.ts:87]: yeah, I had just AI generated their descriptions back on the branch but they don't have their own sprites, etc.

File: convex/agent/memory.ts (hunk 5 above)
Comment (sujayakar) [reply convex/agent/memory.ts:300]: we clear the flag within `insertMemory` to have it be atomic w/inserting the memory -- I'll add a comment here.

File: convex/world.ts (hunk 2 above)
Comment (sujayakar) [reply convex/world.ts:246]: not yet! yeah, the conversations are pretty fast paced atm -- filed https://linear.app/convex/issue/CX-5035/add-a-wpm-limit-to-agent-messages to think of it as a "WPM limit"

**Pull request merged**